src/Api/AdminConsole/Controllers/OrganizationUsersController.cs

```diff
@@ -209,32 +210,30 @@ private async Task<ListResponseModel<OrganizationUserUserDetailsResponseModel>>
         var readAllAuthorized = (await _authorizationService.AuthorizeAsync(
             User, new OrganizationScope(orgId), OrganizationUserUserDetailsOperations.ReadAll)).Succeeded;
 
-        if (!readAllAuthorized)
+        if (readAllAuthorized)
         {
-            var readWithAccountRecovery = (await _authorizationService.AuthorizeAsync(
-                User, new OrganizationScope(orgId), OrganizationUserUserDetailsOperations.ReadAccountRecovery)).Succeeded;
+            var results = await _organizationUserUserDetailsQuery.Get(request);
+            var responseList = results
+            .Select(result => new OrganizationUserUserDetailsResponseModel(result.OrgUser, result.TwoFactorEnabled, result.ClaimedByOrganization))
```

Comment:
I'd rather avoid having multiple bools as positional arguments. Please add a new ctor for this model that takes the tuple directly.